src/app.py

```diff
@@ -0,0 +1,139 @@
+    task_names = [
+        "pa",
+        "word",
+        "passage",
+        "sentence_normal",
+        "sentence_fast",
+        "sentence_clear",
+    ]
+    participants_ID = [
+        "01",
+        "02",
+        "03",
+        "04",
+        "05",
+        "06",
+        "07",
+        "08",
+        "09",
+        "10",
+        "11",
+        "12",
+        "13",
+        "14",
+        "15",
+        "16",
+        "17",
+        "18",
+        "19",
+        "20",
+    ]
+    participants_Gender = ["F", "M", "N"]
+    sessions_ID = ["S1", "S2", "S3", "S4", "S5", "S6", "S7", "S8"]
+
+    if data_file:
+        st.subheader("Variables")
+        st.markdown(
+            "Select the desired variable values, then click the 'Submit' button"
+        )
+        with st.form("variables_form"):
+
+            # Add select boxes side by side:
+            col1, col2, col3 = st.columns(3)
+            with col1:
+                talker_ID = st.selectbox(
+                    "Participant ID:", participants_ID, label_visibility="visible"
+                )
+            with col2:
+                talker_G = st.selectbox(
+                    "Participant Gender:",
+                    participants_Gender,
+                    label_visibility="visible",
+                )
+            with col3:
+                session_ID = st.selectbox(
+                    "session ID:", sessions_ID, label_visibility="visible"
+                )
+
+            # Add Submit Button
+            submitted = st.form_submit_button("Submit")
+
+        if submitted:
+            # Repeat the following for every task
+            output_path = "_".join([corpus, talker_ID, talker_G, session_ID])
+            outputs, names = [], []
+            for i in range(len(tasks_ID)):
+                task_df = data[
+                    data["task_name"] == task_names[i]
+                ]  # selects rows according to the task
+                task_df = task_df[
+                    "item_text"
+                ].dropna()  # drop empty rows (first row for every task)
+                task_df.index = (
+                    task_df.reset_index(drop=True).index + 1
+                )  # reset index, starting at 1
+                task_df = (
+                    task_df.index.astype(str) + ". " + task_df
+                )  # add index to text
+
+                # Define output name, based in variables
+                output_name = "_".join(
+                    [corpus, tasks_ID[i], talker_ID, talker_G, session_ID + ".txt"]
+                )
+
+                # Store data and names
+                outputs.append(task_df)
+                names.append(output_name)
+
+            # Transform data into a zip file
+            with ZipFile("output.zip", "w") as csv_zip:
+                # annotation files
+                for i in range(len(outputs)):
+                    csv_zip.writestr(names[i], "\n".join(task_df))
```

Comment:
It is saving the same dataframe for all annotation files, to fix:

```suggestion
                    csv_zip.writestr(names[i], "\n".join(outputs[i]))
```